src/bootstrap/bootstrap.py

```diff
@@ -12,9 +12,25 @@
 import sys
 import tarfile
 import tempfile
+import sqlite3
 
 from time import time
 
+def acquire_lock():
+    try:
+        con = sqlite3.Connection("xlock.db", timeout=0)
+        curs = con.cursor()
+        curs.execute("BEGIN EXCLUSIVE")
+        return curs
+    except sqlite3.OperationalError:
+        del con
+        del curs
+        print("Waiting for lock on database")
```

Comment:
I think database here is an internal implementation detail -- I would suggest a message closer to Cargo's:

>  Blocking waiting for file lock on the registry index

Maybe: "Blocking waiting for build directory lock".